src/Surfnet/StepupSelfService/SelfServiceBundle/Security/Authentication/Session/SessionStorage.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/**
+ * Copyright 2014 SURFnet bv
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\StepupSelfService\SelfServiceBundle\Security\Authentication\Session;
+
+use Surfnet\StepupSelfService\SelfServiceBundle\Exception\LogicException;
+use Surfnet\StepupSelfService\SelfServiceBundle\Security\Authentication\SamlAuthenticationStateHandler;
+use Surfnet\StepupSelfService\SelfServiceBundle\Security\Authentication\AuthenticatedSessionStateHandler;
+use Surfnet\StepupSelfService\SelfServiceBundle\Value\DateTime;
+use Symfony\Component\HttpFoundation\Session\SessionInterface;
+
+class SessionStorage implements AuthenticatedSessionStateHandler, SamlAuthenticationStateHandler
+{
+    /**
+     * Session keys
+     */
+    const AUTH_SESSION_KEY = '__auth/';
+    const SAML_SESSION_KEY = '__saml/';
+
+    /**
+     * @var \Symfony\Component\HttpFoundation\Session\SessionInterface
+     */
+    private $session;
+
+    /**
+     * @param SessionInterface $session
+     */
+    public function __construct(SessionInterface $session)
+    {
+        $this->session = $session;
+    }
+
+    public function logAuthenticationMoment()
```

Comment:
Other than in tests, I don't see this being called anywhere.

I think this should be done upon (first) authentication in order to check whether the session lifetime exceeds the absolute limit or not.